core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPushMergeWriterUpdateIntoConnector.java

```diff
@@ -0,0 +1,185 @@
+                        Symbol mergeRow = p.symbol("merge_row");
+                        Symbol rowId = p.symbol("row_id");
+                        Symbol rowCount = p.symbol("row_count");
+                        // set arithmetic expression which we don't support yet
+                        Expression updateMergeRowExpression = new Row(ImmutableList.of(p.symbol("col1").toSymbolReference(),
+                                new ArithmeticBinaryExpression(ArithmeticBinaryExpression.Operator.MULTIPLY, p.symbol("col1").toSymbolReference(), new LongLiteral("5"))));
+
+                        return p.tableFinish(
+                                p.merge(
+                                        p.mergeProcessor(SCHEMA_TABLE_NAME,
+                                                p.project(new Assignments(Map.of(mergeRow, updateMergeRowExpression)),
+                                                        p.tableScan(tableScanBuilder -> tableScanBuilder
+                                                                .setAssignments(ImmutableMap.of())
+                                                                .setSymbols(ImmutableList.of())
+                                                                .setTableHandle(ruleTester.getCurrentCatalogTableHandle(TEST_SCHEMA, TEST_TABLE)).build())),
+                                                mergeRow,
+                                                rowId,
+                                                ImmutableList.of(),
+                                                ImmutableList.of(),
+                                                ImmutableList.of()),
+                                        p.mergeTarget(SCHEMA_TABLE_NAME, new TableWriterNode.MergeParadigmAndTypes(Optional.of(DELETE_ROW_AND_INSERT_ROW), ImmutableList.of(), columnNames, INTEGER)),
+                                        mergeRow,
+                                        rowId,
+                                        ImmutableList.of()),
+                                p.mergeTarget(SCHEMA_TABLE_NAME),
+                                rowCount);
+                    })
+                    .doesNotFire();
+        }
+    }
+
+    @Test
+    public void testPushUpdateIntoConnectorUpdateAll()
+    {
+        List<String> columnNames = ImmutableList.of("col1", "col2");
+        MockConnectorFactory factory = MockConnectorFactory.builder().build();
+        try (RuleTester ruleTester = RuleTester.builder().withDefaultCatalogConnectorFactory(factory).build()) {
+            ruleTester.assertThat(createRule(ruleTester))
+                    .on(p -> {
+                        Symbol mergeRow = p.symbol("merge_row");
+                        Symbol rowId = p.symbol("row_id");
+                        Symbol rowCount = p.symbol("row_count");
+                        // set function call, which represents update all columns statement
+                        // TODO modify, when we add support for update all
```

Comment:
Remove TODO, it is obvious that we need to update this test if support this case